### ISSUE TITLE
Insert error when column type is char(0)

### ISSUE THREAD

**Issue by jackysp** [pingcap/tidb#3832]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```sql
create table t1 (b char(0));
insert into t1 values ("");
```

2. What did you expect to see?

```
mysql> insert into t1 values ("");
Query OK, 1 row affected (0.00 sec)
```

3. What did you see instead?

```
mysql> insert into t1 values ("");
ERROR 1406 (22001): Data Too Long, field len 0, data len 0
```

4. What version of TiDB are you using (`tidb-server -V`)?

**Comment by shenli** [comment]
Should we close this issue? @jackysp

**Comment by jackysp** [comment]
Fixed #3849
